pkg/cmd/util/flag/choices.go

```diff
@@ -0,0 +1,42 @@
+package flag
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+type Choices struct {
+	Choices []string
+	Value   string
+}
+
+// NewChoices represents a string flag with restricted values
+func NewChoices(choices []string, def string) Choices {
+	return Choices{
+		Choices: choices,
+		Value:   def,
+	}
+}
+
+func (c Choices) String() string {
+	return c.Value
+}
```

Comment:
In keeping with the way pflag is written, a pointer receiver would be better here.

https://github.com/spf13/pflag/blob/2e9d26c8c37aae03e3f9d4e90b7116f5accb7cab/string.go#L19
```suggestion
func (c *Choices) String() string {
	return c.Value
}
```